UI/MapUI.gd

```diff
@@ -18,6 +24,59 @@ func _input(_inputevent: InputEvent) -> void:
 
 func _on_return_button_press():
 	queue_free()
+	
+
+func _on_add_torch_pressed():
+	torch_confirmation_dialog.show()
+
+func _close_torch_placement_dialog():
+	torch_confirmation_dialog.hide()
+
+func _add_torch_to_current_location():
+	var current_player_floor_index = current_player_room.floor_index
+	var current_player_room_index = current_player_room.room_index
+	
+	var room_light_range = 1
+	var max_light_index = current_player_floor_index + _LIGHT_FLOOR_RANGE + 1 
+	
+	if current_player_floor_index + _LIGHT_FLOOR_RANGE + 1 > MapManager.map_width_array.size():
```

Comment:
isn't that already `max_light_index` ? no need to recalculate it

Reply:
Everything in `_add_torch_to_current_location()` refactored to use `MapMovement.get_all_accessible_rooms_in_range()`